azure-pipelines-official.yml

```diff
@@ -37,6 +38,30 @@ stages:
     - powershell: Write-Host "##vso[task.setvariable variable=SourceBranchName]$('$(Build.SourceBranch)'.Substring('refs/heads/'.Length))"
       displayName: Setting SourceBranchName variable
 
+    - powershell: Write-Host "##vso[task.setvariable variable=IsPrValidationBuild]($(PrNumber) -ne '') -and ($(PrNumber) -ne 'default')"
+      displayName: Setting IsPrValidationBuild variable
+
+    - task: tagBuildOrRelease@0
+      displayName: Tag build
+      inputs:
+        type: 'Build'
+        tags: if (variables['IsPrValidationBuild']) { 'PrValidationBuild' } else { "OfficialBuild" }
+
+    - powershell: |
+        if ($SourceBranchName -notlike '*-vs-deps') {
```

Comment:
>if ($SourceBranchName -notlike '*-vs-deps') { [](start = 8, length = 45)

Could you extract this to a ps1 file in eng dir? That way one can run it manually e.g. for testing purposes

Reply:
Sure, will do.